### PR TITLE
fix dev code for webpack compatibility

### PR DESCRIPTION
**Pull request by cherifGsoul**
This fixes the first part of this issue canjs/canjs#4170 to remove debug code from the production builds.

### CODE REVIEW

File: src/hashchange.js [hunk 1]
```diff
@@ -42,14 +42,18 @@ canReflect.assign(HashchangeObservable.prototype,{
...
-            queues.enqueueByQueue(this.handlers.getNode([]), this, [this.value, old]
-                //!steal-remove-start
-                /* jshint laxcomma: true */
-                , null
-                , [ canReflect.getName(this), "changed to", this.value, "from", old ]
-                /* jshint laxcomma: false */
-                //!steal-remove-end
-            );
+        	if(process.env.NODE_ENV !== 'production') {
+	            queues.enqueueByQueue(this.handlers.getNode([]), this, [this.value, old]
```
Comment (matthewp) [inline src/hashchange.js]: This still needs to be called in production. It's just the arguments in the remove-start/stop that we don't want in prod.

File: src/hashchange.js [hunk 2]
```diff
@@ -67,20 +71,25 @@ canReflect.assign(HashchangeObservable.prototype,{
...
 		return true;
-	},
-	//!steal-remove-start
-	"can.getName": function() {
+	}
+};
+
+//!steal-remove-start
+if(process.env.NODE_ENV !== 'production') {
+	hashchangeObservableProto["can.getName"] = function() {
```
Comment (matthewp) [inline src/hashchange.js]: This should be `hashchangeObservableProto[canSymbol.for("can.getName")]`

File: src/hashchange.js [hunk 3]
```diff
@@ -12,63 +13,65 @@ var SimpleObservable = require("can-simple-observable");
...
+			//!steal-remove-start
+			if(process.env.NODE_ENV !== 'production') {
+				reasonLog = [ canReflect.getName(this), "changed to", this.value, "from", old ];
+			}
+			//!steal-remove-end
+			
+			
+			queues.enqueueByQueue(this.handlers.getNode([]), this, [this.value, old]
+				/* jshint laxcomma: true */
+				, null
```
Comment (matthewp) [inline src/hashchange.js]: This null argument should only be passed in dev mode.

Comment (cherifGsoul) [reply src/hashchange.js]: Fixed!

**Pull request merged**